src/components/tabs/Tabs.vue

```diff
@@ -12,27 +12,59 @@
 </template>
 
 <script>
-  import {createUniqueId} from '../../utils'
-
   export default {
+    data() {
+      return {
+        tabs: []
+      }
+    },
     mounted() {
-      const link = AJS.$(this.$el).find('.menu-item a')[0];
-      const $link = AJS.$(link);
-      AJS.tabs.change($link)
-      AJS.tabs.setup();
+      this.updateTabs()
+      AJS.tabs.setup()
     },
-
-    computed: {
-      tabs() {
-        const tabs = this.$slots.default.filter(slot => slot.componentOptions
-          && (slot.componentOptions.tag === "va-tab" || slot.componentOptions.tag === "aui-tab"));
-        for (let obj of tabs) {
-          if (!obj.tab_id) {
-            obj.tab_id = createUniqueId('tab')
-            obj.componentOptions.propsData.tabId = obj.tab_id
-          }
+    updated(){
+      this.updateTabs()
+    },
+    methods: {
+      selectDefaultTab() {
+        if (this.tabs.length){
+          this.selectTab(this.tabs[0].hash)
         }
+      },
+      hasActiveTab(){
+        let activeTab = $('.menu-item.active-tab')
+        return activeTab.length && activeTab.is(':visible')
+      },
+      selectTab(tabHash) {
+        Vue.nextTick(() => {
+          const tabLinks = this.$refs[`tab_link_${tabHash}`]
+          if (tabLinks && tabLinks.length) {
+            let $tabLink = $(tabLinks[0])
+            AJS.tabs.change($tabLink)
+          }
+        })
+      },
+      getTabs() {
+        let tabs = []
+        this.$slots.default.forEach(vNode => {
+          if (vNode.componentInstance && vNode.componentOptions.tag === "va-tab") {
+            let tab = vNode.componentInstance
+            tabs.push(tab)
+          }
+        })
         return tabs
```

Comment:
You can replace imperative array creation with a simple `filter + map`. Not tested but sth similar to this:

      getTabs() {
        return this.$slots.default
          .filter(vNode => vNode.componentInstance && vNode.componentOptions.tag === "va-tab")
          .map(vNode => vNode.componentInstance);
      },

Reply:
Please check my last commit a6357b2